### PR TITLE
Plugins: Redirect to Plugin List from Settings screen

### PR DESCRIPTION
**Pull request by itsmeichigo**
Part of #4114 and #3921 

## Description
To display the list of installed plugins for a selected site, a new view controller needs to be created and showed from the Plugins entry point in the Settings screen. 

## Solution
- I chose MVVM pattern and UIKit to build the Plugin List screen since there's no native support for refresh control in SwiftUI and current solutions aren't really performant and straightforward.
- The view model is injected into PluginListViewController upon initialization. Since I added the view controller to the Dashboard storyboard, I also updated the storyboard extension to [accept `creator` block](https://developer.apple.com/documentation/uikit/uistoryboard/3173166-instantiateviewcontrollerwithide?language=objc), which is a good place for dependency injection.
- The view model is incomplete, please ignore it for now.

## Testing
When tapping on the Plugins entry point on the Settings screen, a new screen should be pushed. This screen is currently empty and will be updated in the next PR.

## Note
I left a `TODO` where the tap on Plugins entry point is handled - I'm not sure if we need to track this tap action. Since every other actions in the Settings screen is tracked, I think it makes sense to do that for Plugins also. Let me know if this is necessary and if it is, I'd like to know the name of the event that needs to be tracked.

Update release notes:
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/Extensions/UIStoryboard+Woo.swift [hunk 1]
```diff
@@ -23,9 +23,10 @@ extension UIStoryboard {
...
     /// Returns a view controller from a Storyboard assuming the identifier is the same as the class name.
+    /// Accepts an optional creator to inject additional dependencies.
     ///
-    func instantiateViewController<T: NSObject>(ofClass classType: T.Type) -> T? {
+    func instantiateViewController<T: UIViewController>(ofClass classType: T.Type, creator: ((NSCoder) -> T?)? = nil) -> T {
         let identifier = classType.classNameWithoutNamespaces
-        return instantiateViewController(withIdentifier: identifier) as? T
+        return instantiateViewController(identifier: identifier, creator: creator)
     }
 }
```
Comment (Ecarrion) [inline WooCommerce/Classes/Extensions/UIStoryboard+Woo.swift]: Hey @itsmeichigo, we have been actively trying to migrate away from storyboards. Probably our fault if this is not well documented :-(

Could you please re-work it to use Xibs or Code?

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Plugins/PluginListViewController.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import UIKit
+
+class PluginListViewController: UIViewController {
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Plugins/PluginListViewController.swift]: Nit: This can be `final`

Comment (Ecarrion) [reply WooCommerce/Classes/ViewRelated/Dashboard/Settings/Plugins/PluginListViewController.swift]: Also, could you add a quick description of the class?

File: WooCommerce/Classes/ViewRelated/Dashboard/Settings/Plugins/PluginListViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+final class PluginListViewModel {
+
+    /// Reference to the StoresManager to dispatch Yosemite Actions.
+    ///
+    private let storesManager: StoresManager
+
+    init(storesManager: StoresManager = ServiceLocator.stores) {
+        self.storesManager = storesManager
+    }
```
Comment (Ecarrion) [inline WooCommerce/Classes/ViewRelated/Dashboard/Settings/Plugins/PluginListViewModel.swift:12]: +1 for setting things up for dependency injection!

**Review by Ecarrion — CHANGES_REQUESTED** [review]
I think you forgot to merge develop back onto this branch because the `assertionFailure` on `Dispatcher` is still present. 😅

**Comment by itsmeichigo** [comment]
Thanks @Ecarrion, I have merged changes from develop and migrated the view controller to use xib instead!

**Review by Ecarrion — APPROVED** [review]
Looks good to me! Thanks for changing from Storyboards to Xibs.

**Pull request merged**